auditbeat/module/auditd/audit_linux.go

```diff
@@ -104,19 +112,37 @@ func (ms *MetricSet) Run(reporter mb.PushReporterV2) {
 		return
 	}
 
-	out, err := ms.receiveEvents(reporter.Done())
+	out, statusC, err := ms.receiveEvents(reporter.Done())
 	if err != nil {
 		reporter.Error(err)
 		ms.log.Errorw("Failure receiving audit events", "error", err)
 		return
 	}
 
+	var lostUpdateTimer <-chan time.Time
+	if ms.kernelLost.enabled {
+		timer := time.NewTicker(lostEventsUpdateInterval)
+		defer timer.Stop()
+		lostUpdateTimer = timer.C
+	}
+
 	for {
 		select {
 		case <-reporter.Done():
 			return
+
 		case msgs := <-out:
 			reporter.Event(buildMetricbeatEvent(msgs, ms.config))
+
+		case status := <-statusC:
```

Comment:
Reading from `statusC` and sending the status request should probably be done from its own goroutine because when the output is blocked due to back-pressure (e.g. `reporter.Event()` is slow) then no metrics are updated.